### ISSUE TITLE
As a webcomponent store visitor, I would prefer a lightweight home page that doesn't download all webcomponents (~15Mb) on every visit.

### ISSUE THREAD

**Issue by dulvui** [noi-techpark/it.bz.opendatahub.webcomponents#213]
The new preview feature in the main page listing causes the webcomponent store to download every single webcomponent to show its preview.
The issue with that is that some webcomponents are quite heavy and can reach also some megabytes in size and currently on every visit ~15Mb of data gets downloaded. This also slows down the whole homepage and will only get worse with every new webcomponent on the store.

A possible solutions could be to load a specific webcomponent, only when the preview button is pressed.

**Comment by dulvui** [comment]
cc @sseppi @mrabans

**Comment by mrabans** [comment]
@dulvui absolutely, this is a bug. It should be already the case. I'll handle this.

**Comment by sseppi** [comment]
@dulvui I agree with @mrabans , it was already planned to be as you proposed.

Martin, will you communicate it to Marco and assign him the issue, did I got it right?

**Comment by mrabans** [comment]
@sseppi exaclty

**Comment by dallefeste** [comment]
@dulvui, thanks for your report. I've investigated the issue and can confirm that the web components are loaded only after clicking the 'Preview' button. 

However, I found that the problem is related to the resolution of the loaded preview images. The thumbnail images currently do not have a sufficiently high resolution as required. We can address this issue in several alternative ways:
- Switch to the thumbnail images and adjust their resolutions in each web component repository;
- Keep full-size images and optimize them in each web component repository;
- Use an image provider service, if available.

The specific code line I'm referring to is line number 303 of src/components/webcomponent-entry-card.vue file.

**Comment by dulvui** [comment]
@dallefeste You are right, I didn't see that the images are the problem. I can confirm that the webcomponents are loaded only when preview is pressed.
I'll check if the API has already a thumbnail functionality or something similar.

**Comment by dulvui** [comment]
@dallefeste I checked now and the thumbnails are currently set to width=400 height=200.
What resolution would you need?

**Comment by dallefeste** [comment]
@dulvui Thanks for your support. It should be okay with width=600 height=400. I will proceed with a new PR, where the preview switches to the /thumb URL.

**Comment by dulvui** [comment]
@dallefeste I already changed everything and now the thumbnail size now 800x600 and it seems okay. So no PR from your side is needed. It can be configured in the environment, so now it ca be adapted easily.

It is already live on testing https://webcomponents.opendatahub.testingmachine.eu
Let me know if this resolution is okay, then I'll put it to production.

**Comment by mrabans** [comment]
@dulvui @dallefeste looks good with this resolution.

**Comment by dulvui** [comment]
It is live on production now and the site load is now ~4Mb.
